fendermint/eth/api/src/gas/mod.rs

```diff
@@ -68,8 +68,27 @@ pub fn median_gas_premium(prices: &mut [(TokenAmount, i64)], block_gas_target: i
 
     if prev2 != TokenAmount::zero() {
         premium += &prev2;
-        premium.div_ceil(BigInt::from(2));
+        premium.div_ceil(BigInt::from(2))
+    } else {
+        premium
     }
```

Comment:
I haven't reviewed this full method and its intended usage in depth (and it looks a bit unidiomatic). But there does indeed seem to be a logical error here. That said, this looks more idiomatic?

```rust
let premium = if !prev2.is_zero() {
    (prev1 + &prev2).div_ceil(BigInt::from(2))
} else {
    prev1
}
premium
```